nesta/core/routines/health_data/nih_data/mesh_join_task.py

```diff
@@ -0,0 +1,118 @@
+    test = luigi.BoolParameter()
+
+    @staticmethod
+    def format_mesh_terms(df):
+        """
+        Removes unrequired columns and pivots the mesh terms data into a dictionary.
+
+        Args:
+            df (dataframe): mesh terms as returned from retrieve_mesh_terms
+
+        Returns:
+            (dict): document_id: list of mesh terms
+        """
+        logging.info("Formatting mesh terms")
+        # remove PRC rows
+        df = df.drop(df[df.term == 'PRC'].index, axis=0)
+
+        # remove invalid error rows
+        df = df.drop(df[df.doc_id.astype(str).str.contains('ERROR.*ERROR', na=False)].index, axis=0)
+        df['term_id'] = df['term_id'].apply(lambda x: int(x[1:]))
+
+        # pivot and remove unrequired columns
+        doc_terms = {
+            doc_id: {'terms': grouped.term.values, 'ids': grouped.term_id.values}
+            for doc_id, grouped in df.groupby("doc_id")}
+        return doc_terms
+    
+    @staticmethod
+    def get_abstract_file_keys(bucket, key_prefix):
+        s3 = boto3.resource('s3')
+        s3bucket = s3.Bucket(bucket)
+        return {o.key for o in s3bucket.objects.filter(Prefix=key_prefix)}
+
+    def output(self):
+        db_config = get_config(os.environ[self.db_config_env], "mysqldb")
+        db_config['database'] = 'dev' if self.test else 'production'
+        db_config['table'] = "MeshTerms <dummy>"
+        update_id = "NihJoinMeshTerms_{}".format(self.date)
+        return MySqlTarget(update_id=update_id, **db_config)
+
+    def run(self):
+        db = 'production' if not self.test else 'dev'
+
+        bucket = 'innovation-mapping-general'
+        key_prefix = 'nih_abstracts_processed/mti'
+        keys = self.get_abstract_file_keys(bucket, key_prefix)
+        
+        engine = get_mysql_engine(self.db_config_env, 'mysqldb', db)
+        with db_session(engine) as session:
+            
+            if self.test:
+                existing_projects = {int(p.application_id) for p in
+                        session.query(Projects.application_id).distinct()}
+
+            projects_done = {int(p.project_id) 
+                for p in session.query(ProjectMeshTerms.project_id).distinct()}
+            
+            mesh_term_ids = {int(m.id) for m in session.query(MeshTerms.id).all()}
+
+            logging.info('Inserting associations')
+            
+            for key_count, key in enumerate(keys):
+                if self.test & (key_count > 2):
+                    continue
+                df_mesh = retrieve_mesh_terms(bucket, key)
+                doc_terms = self.format_mesh_terms(df_mesh)
+                data = []
+                for doc_count, (doc, t) in enumerate(doc_terms.items()):
+                    doc_terms = []
+                    if self.test & (doc_count > 2):
+                        continue
+                    if (doc in projects_done) | (doc not in existing_projects):
+                        continue
+                    else:
```

Comment:
`else` isn't required since previous is `continue`: this means you can unindent by one tab to improve readability